library/ssl_cache.c

```diff
@@ -219,12 +219,13 @@ static int ssl_cache_pick_writing_slot(mbedtls_ssl_cache_context *cache,
     if (old == NULL) {
         /* This should only happen on an ill-configured cache
          * with max_entries == 0. */
-        return 1;
+        return MBEDTLS_ERR_SSL_INTERNAL_ERROR;
```

Comment:
Should we change to `MBEDTLS_ERR_SSL_ILLEGAL_PARAMETER`? AFAIU, this only happens when user doesn't set `max_entries` but still call `mbedtls_ssl_cache_set` to set cache entry for a session. It seems like a illegal configuration.

Reply:
I think we should not. See: https://github.com/Mbed-TLS/mbedtls/blob/ac6db4d649d9c0ec086b57438fddbdf56482393f/include/mbedtls/ssl.h#L153-L154

It is used to indicate an illegal parameters in ssl/tls messages, not the case here. Apart from that, `max_entries = 0` is a valid configuration, the thing is the user unexpectlly calls `mbedtls_ssl_cache_set` when `max_entries = 0`. So, I think `MBEDTLS_ERR_SSL_INTERNAL_ERROR` is enough.